core/bones/color.py

```diff
@@ -11,13 +11,15 @@ def __init__(self, *, mode="rgb", **kwargs):  # mode rgb/rgba
         assert mode in {"rgb", "rgba"}
         self.mode = mode
 
-    def singleValueFromClient(self, value, skel: 'viur.core.skeleton.SkeletonInstance', name: str, origData):
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.